backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/questionsubmission/QuestionSubmissionService.java

```diff
@@ -0,0 +1,278 @@
+
+    @Retryable(value = {SQLException.class}, backoff = @Backoff(delay = 5000))
+    @Transactional(isolation = Isolation.REPEATABLE_READ)
+    public QuestionSubmissionDto updateQuestionSubmission(Integer questionSubmissionId, QuestionSubmissionDto questionSubmissionDto) {
+        QuestionSubmission questionSubmission = getQuestionSubmission(questionSubmissionId);
+
+        User user = getStudent(questionSubmissionDto.getUserId());
+
+        if(user.isStudent() && questionSubmission.getQuestion().getStatus() != Question.Status.IN_REVISION) {
+            throw new TutorException(CANNOT_EDIT_REVIEWED_QUESTION);
+        }
+
+        this.questionService.updateQuestion(questionSubmissionDto.getQuestion().getId(), questionSubmissionDto.getQuestion());
+        return new QuestionSubmissionDto(questionSubmission);
+    }
+
+    @Retryable(
+            value = { SQLException.class },
+            backoff = @Backoff(delay = 5000))
+    @Transactional(isolation = Isolation.READ_COMMITTED)
+    public void updateQuestionSubmissionTopics(Integer questionSubmissionId, TopicDto[] topics) {
+        Integer questionId = questionSubmissionRepository.findQuestionIdByQuestionSubmissionId(questionSubmissionId);
+
+        questionService.updateQuestionTopics(questionId, topics);
+    }
+
+    @Retryable(
+            value = { SQLException.class },
+            backoff = @Backoff(delay = 5000))
+    @Transactional(isolation = Isolation.READ_COMMITTED)
+    public void removeSubmittedQuestion(Integer questionSubmissionId) {
+        QuestionSubmission questionSubmission = questionSubmissionRepository.findById(questionSubmissionId).orElseThrow(() -> new TutorException(QUESTION_SUBMISSION_NOT_FOUND, questionSubmissionId));
+        Question question = questionRepository.findById(questionSubmission.getQuestion().getId()).orElseThrow(() -> new TutorException(QUESTION_NOT_FOUND, questionSubmission.getQuestion().getId()));
+
+        if(!questionSubmission.getReviews().isEmpty() || !question.getStatus().equals(Question.Status.IN_REVISION)) {
+            throw new TutorException(CANNOT_DELETE_REVIEWED_QUESTION);
+        } else {
+            removeQuestionSubmission(questionSubmission.getId());
+            question.remove();
+            questionRepository.delete(question);
+        }
+    }
+
+    @Retryable(value = {SQLException.class}, backoff = @Backoff(delay = 5000))
+    @Transactional(isolation = Isolation.REPEATABLE_READ)
+    public void removeQuestionSubmission(Integer questionSubmissionId) {
+        QuestionSubmission questionSubmission = getQuestionSubmission(questionSubmissionId);
+
+        removeReviews(questionSubmission);
+
+        questionSubmission.remove();
+        questionSubmissionRepository.delete(questionSubmission);
+    }
+
+    @Retryable(value = {SQLException.class}, backoff = @Backoff(delay = 5000))
+    @Transactional(isolation = Isolation.REPEATABLE_READ)
+    public void removeReviews(QuestionSubmission questionSubmission) {
```

Comment:
should this be a service, do we have removeReviews in the UI? or only remove question submission?

Reply:
good point, it is only refactored code from the previous method an should therefore not be a service since there is no way to access it in the ui